metaserver/src/main/scala/scala/meta/languageserver/Linter.scala

```diff
@@ -71,27 +71,22 @@ class Linter(
       val configDecoder = ScalafixReflect.fromLazySemanticdbIndex(lazyIndex)
       val (rule, config) =
         ScalafixConfig.fromInput(configInput, lazyIndex)(configDecoder).get
-      val results: Seq[PublishDiagnostics] = index.database.documents.map { d =>
-        val tree = Parser.parse(d).get
-        val ctx = RuleCtx.applyInternal(tree, config)
-        val patches = rule.fixWithNameInternal(ctx)
-        val diagnostics =
-          Patch.lintMessagesInternal(patches, ctx).map(toDiagnostic)
-        val uri = d.input.syntax
-        PublishDiagnostics(uri, diagnostics)
+      val results: Seq[PublishDiagnostics] = index.database.documents.flatMap { d =>
+        Parser.parse(d).toOption.map { tree =>
```

Comment:
we can't ignore parse errors anymore, since this can be invoked while the user is typing.